recipe/patches/0007-add-PROTOBUF_EXPORT-for-google-protobuf-io-SafeDoubl.patch

```diff
@@ -0,0 +1,23 @@
+From 16c6f389a0b38196b89add737dc3a6337e534273 Mon Sep 17 00:00:00 2001
+From: "H. Vetinari" <h.vetinari@gmx.com>
+Date: Wed, 31 May 2023 08:31:06 +1100
+Subject: [PATCH 7/7] add PROTOBUF_EXPORT for
+ google::protobuf::io::SafeDoubleToFloat
+
+---
+ src/google/protobuf/io/strtod.h | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/src/google/protobuf/io/strtod.h b/src/google/protobuf/io/strtod.h
+index 851c8e621..648f1ff1c 100644
+--- a/src/google/protobuf/io/strtod.h
++++ b/src/google/protobuf/io/strtod.h
+@@ -65,7 +65,7 @@ double NoLocaleStrtod(const char* str, char** endptr);
+ // Casts a double value to a float value. If the value is outside of the
+ // representable range of float, it will be converted to positive or negative
+ // infinity.
+-float SafeDoubleToFloat(double value);
++PROTOBUF_EXPORT float SafeDoubleToFloat(double value);
```

Comment:
Interestingly, adding `PROTOBUF_EXPORT` does not seem to be enough to make this symbol visible (even though other symbols in that TU already use that macro).

Reply:
Ah, it actually [needs](https://github.com/protocolbuffers/protobuf/blob/v23.2/src/google/protobuf/port_def.inc#L523-L526) `PROTOBUF_USE_DLLS` after all (which I find very confusing as an option on unix TBH...)

Reply:
Ah no, I need to actually rebuild the lib with the `PROTOBUF_EXPORT` patch... 🤦